src/xtgeo/xyz/_xyz.py

```diff
@@ -271,7 +273,36 @@ def get_boundary(self):
 
         return (xmin, xmax, ymin, ymax, zmin, zmax)
 
-    def operation_polygons(self, poly, value, opname="add", inside=True):
+    def mark_in_polygons(
+        self,
+        poly: Union["Polygons", List["Polygons"]],  # noqa: F821
+        name: str = "pstatus",
+        inside_value: int = 1,
+        outside_value: int = 0,
+    ):
+        """Add a column that assign values if points are inside or outside polygons.
+
+        This is a generic function that adds a column in the points dataframe with
+        a flag for values being inside or outside polygons in a Polygons instance.
+
+        Args:
+            poly: One single xtgeo Polgons instance, or a list of Polygons instances.
+            name: Name of column that flags inside or outside status
+            inside_value: Flag value for being inside polygons
+            outside_value: Flag value for being outside polygons
+
+        ..versionadded:: 3.2
+        """
+        _xyz_oper.mark_in_polygons_mpl(self, poly, name, inside_value, outside_value)
+
+    def operation_polygons(
+        self,
+        poly: Union["Polygons", List["Polygons"]],  # noqa: F821
+        value: float,
+        opname: str = "add",
```

Comment:
I've no clue the how deep/exposed the opname semantics are, but maybe a nice change in the future would be a str enum to hold the operation types as constants rather than magic strings

Reply:
Yes, a good idea. It is noted - perhaps have as a separate PR